cmd/system-probe/config/adjust_npm.go

```diff
@@ -0,0 +1,76 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package config
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+)
+
+const (
+	maxConnsMessageBatchSize     = 1000
+	maxOffsetThreshold           = 3000
+	defaultMaxProcessesTracked   = 1024
+	defaultMaxTrackedConnections = 65536
+)
+
+func adjustNetwork(cfg config.Config) {
+	limitMaxInt(cfg, spNS("max_conns_per_message"), maxConnsMessageBatchSize)
+
+	if cfg.GetBool(spNS("disable_tcp")) {
+		cfg.Set(netNS("collect_tcp_v4"), false)
+		cfg.Set(netNS("collect_tcp_v6"), false)
+	}
+	if cfg.GetBool(spNS("disable_udp")) {
+		cfg.Set(netNS("collect_udp_v4"), false)
+		cfg.Set(netNS("collect_udp_v6"), false)
+	}
+	if cfg.GetBool(spNS("disable_ipv6")) || !kernel.IsIPv6Enabled() {
+		cfg.Set(netNS("collect_tcp_v6"), false)
+		cfg.Set(netNS("collect_udp_v6"), false)
+	}
+
+	if runtime.GOOS == "windows" {
+		validateInt(cfg, spNS("closed_connection_flush_threshold"), 0, func(v int) error {
+			if v < 1024 {
+				return fmt.Errorf("closed connection notification threshold set to invalid value %d. resetting to default", v)
+			}
+			return nil
+		})
+	}
+
+	validateInt64(cfg, spNS("max_tracked_connections"), defaultMaxTrackedConnections, func(v int64) error {
```

Comment:
Shouldn't this be something like `MaxInt` or `MaxUint32`?

Reply:
`Int64` is the type, not the valid range.

Reply:
I can add a check for the maximum too, if you like.

Reply:
Yes that is what I meant.

Reply:
Added